### ISSUE TITLE
update tarfile behavior

### ISSUE THREAD

**Issue by orbeckst** [MDAnalysis/MDAnalysisData#74]
Running the tests in py 3.12 I got the warning:
```
~/anaconda3/envs/py312/lib/python3.12/tarfile.py:2220: DeprecationWarning: Python 3.14 will, by default, filter extracted tar archives and reject files or modify their metadata. Use the filter argument to control this behavior.
    warnings.warn(
```
We need to look into what changes and how we need to adapt the current code (possibly just say "extract all files, I know they are ok").

This may be related to security risks mentioned in PR #58.

**Comment by orbeckst** [comment]
I don't get this warning under 3.14 anymore.
